fml/platform/darwin/string_range_sanitization_unittests.mm

```diff
@@ -28,3 +28,7 @@
   EXPECT_EQ(result.location, 0UL);
   EXPECT_EQ(result.length, 0UL);
 }
+
+TEST(StringRangeSanitizationTest, HandlesEndOfRange) {
+  EXPECT_EQ(fml::RangeForCharacterAtIndex(@"1234", 4).location, 4UL);
```

Comment:
please add assert for the NSRange.length too

Reply:
Done